src/object.jl

```diff
@@ -0,0 +1,35 @@
+import CloudBase: AbstractStore, CloudCredentials
+
+struct Object{T <: AbstractStore, C <: Union{CloudCredentials, Nothing}}
+    store::T
```

Comment:
i'm guessing it might be worth `store` being concretely typed, since we're going to be dispatching on that often? I wonder if we should add conveniences like `const S3Object = Object{AWS.Bucket}`?